OMCompiler/Compiler/FrontEnd/InstExtends.mo

```diff
@@ -95,7 +96,35 @@ protected function instExtendsList
   output list<SCode.AlgorithmSection> outNormalAlgs = {};
   output list<SCode.AlgorithmSection> outInitialAlgs = {};
   output list<SCode.Comment> outComments = {};
+protected
+  list<SCode.Element> duplicates = {};
+  list<String> duplicateUnparseStrings = {};
 algorithm
+  /*
+    Variables can be duplicated after inheritance not before.
+  */
+  duplicates := List.sortedDuplicates(List.sort(inElementsFromExtendsScope,
+                                      SCodeUtil.elementEqual), SCodeUtil.elementEqual);
+  duplicateUnparseStrings := list(SCodeDump.unparseElementStr(i) for i in duplicates);
```

Comment:
A minor comment. Why dump the string here? You can do it in later if you have to actually print it. Instead of checking `listLength(duplicateUnparseStrings) > 1`  you can check for `listLength(duplicates) > 1` and then dump the string if that is satisfied. 
does not make much difference in performance or anything but it is confusing at first glance when you check the string list size instead of the size of the duplicates list.

Reply:
That is true, performance-wise it is not very wise at all to do it up there.. 

Great! Will adjust